xarray/tests/test_backends_datatree.py

```diff
@@ -414,16 +411,30 @@ def test_to_zarr(self, tmpdir, simple_datatree):
     def test_zarr_encoding(self, tmpdir, simple_datatree):
         from numcodecs.blosc import Blosc
 
-        filepath = tmpdir / "test.zarr"
+        filepath = str(tmpdir / "test.zarr")
         original_dt = simple_datatree
 
-        comp = {"compressor": Blosc(cname="zstd", clevel=3, shuffle=2)}
+        blosc = Blosc(cname="zstd", clevel=3, shuffle="shuffle").get_config()
+        comp = {"compressor": {"name": blosc.pop("id"), "configuration": blosc}}
         enc = {"/set2": {var: comp for var in original_dt["/set2"].dataset.data_vars}}
         original_dt.to_zarr(filepath, encoding=enc)
 
         with open_datatree(filepath, engine="zarr") as roundtrip_dt:
-            print(roundtrip_dt["/set2/a"].encoding)
-            assert roundtrip_dt["/set2/a"].encoding["compressor"] == comp["compressor"]
+            retrieved_compressor = roundtrip_dt["/set2/a"].encoding["compressors"][
+                0
+            ]  # Get the BloscCodec object
+            assert (
+                retrieved_compressor.cname.name
+                == comp["compressor"]["configuration"]["cname"]
+            )
+            assert (
+                retrieved_compressor.clevel
+                == comp["compressor"]["configuration"]["clevel"]
+            )
+            assert (
+                retrieved_compressor.shuffle.name
+                == comp["compressor"]["configuration"]["shuffle"]
+            )
```

Comment:
Perhaps we should parameterize this test of zarr_format 2 and 3. Then we can check for the encoding relative to the specified format.

Reply:
I've now managed to ensure that every zarr test is parametrized over format 2/3.